### PR TITLE
Refresh forms for specific price

### PR DESCRIPTION
**Pull request by kompilorb**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Forms in product details, for add special prize are not initialize after validation. Any actualisation on this.  
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-2193
| How to test?  | In BO, catalogs, products. Edit products after go in prizing. Then add a specific prize. Then add the prize, click an other time, the forms must be initialize.

### CODE REVIEW

**Comment by vincentbz** [comment]
Hi, 

thanks !
I just have two boxes when validate the specific price
image
image

But it works, I can create specific prices in a raw without refreshing the page.

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 1]
```diff
@@ -690,7 +702,17 @@ var specificPrices = (function() {
...
-    $('#specific_price_form input').val('');
+    $('#specific_price_form').find('input').each(function () {
+        $(this).val(initSpecificPriceForm[$(this).attr('id')]);
+    });
+    $('#specific_price_form').find('select').each(function () {
+        $(this).val(initSpecificPriceForm[$(this).attr('id')]).change();
+    });
+    $('#specific_price_form').find('input:checkbox').each(function () {
+        alert('5');
+        alert(initSpecificPriceForm[$(this).attr('id')]);
```
Comment (aleeks) [inline admin-dev/themes/default/js/bundle/product/form.js]: Can you remove your trace please ? 😀

**Review by aleeks — CHANGES_REQUESTED** [review]
See comment

**Comment by kompilorb** [comment]
Correction done on incorrect forms popup

**Review by aleeks — APPROVED** [review]

**Comment by aleeks** [comment]
@vincentbz can you check again please ?

File: admin-dev/themes/default/js/bundle/product/form.js [hunk 2]
```diff
@@ -573,6 +573,17 @@ var specificPrices = (function() {
   var productPriceField = $('#form_step2_specific_price_sp_price');
   var discountTypeField = $('#form_step2_specific_price_sp_reduction_type');
   var discountTaxField = $('#form_step2_specific_price_sp_reduction_tax');
+  var initSpecificPriceForm = new Object();
+
+  /** Get all specific prices */
+  function getInitSpecificPriceForm() {
+    $('#specific_price_form').find('select,input').each(function () {
+        initSpecificPriceForm[$(this).attr('id')]=$(this).val();
```
Comment (maximebiloe) [inline admin-dev/themes/default/js/bundle/product/form.js]: I know it's not causing trouble but for the readability could you add spaces around the `=`, please?
It's the same for all the ones you add.

Thank you

**Review by maximebiloe — CHANGES_REQUESTED** [review]

**Comment by vincentbz** [comment]
Tested and ok, thanks !

File: admin-dev/themes/default/js/bundle/product/form.js (hunk 2 above)
Comment (xBorderie) [inline admin-dev/themes/default/js/bundle/product/form.js]: It's even [part of the JS styleguide we use](https://github.com/airbnb/javascript#whitespace--infix-ops) :)

**Comment by kompilorb** [comment]
Add space around "=" done !

**Comment by kompilorb** [comment]
New commit.  Place no space between the argument list and the function name in function calls and declarations.

**Comment by aleeks** [comment]
Thank you @kompilorb

**Pull request merged**